src/thing_creature.c

```diff
@@ -5742,6 +5770,25 @@ void illuminate_creature(struct Thing *creatng)
     lgt->radius <<= 1;
 }
 
+void cleanse_creature(struct Thing* creatng) 
+{
+    if (!creature_is_debuffed(creatng) && creature_affected_by_spell(creatng, SplK_Cleanse)) 
```

Comment:
Why both conditions together here? If the creature is not debuffed, we don't need to cleanse right?